### ISSUE TITLE
custom dataset: 3 classes or one?

### ISSUE THREAD

**Issue by SaadShi** [matterport/Mask_RCNN#2616]
Hey guys,

I'm using mask r CNN to segment the lenses of eyeglasses from selfies. 

I was wondering:  Should I use 3 classes when training (full framed, semi-framed, rimless), since they differ around the lenses, or should I just keep it as one class? 

It would probably work both ways, but what is the best way? Does it matter?
Is there any way to measure which is the best way to do it? 
Should I look at any other metrics apart from IoU?
Is there any training data I could look at while training the model to determine the best way?

Kind regards

**Comment by nataliameira** [comment]
Hi @SaadShi 

I'm a beginner. I have suggestions for you to evaluate:

- 3 classes
- metrics: precision, recall, F1
- evaluate the model while training: use the tensorboard

**Comment by SaadShi** [comment]
> Hi @SaadShi
> 
> I'm a beginner. I have suggestions for you to evaluate:
> 
> * 3 classes
> * metrics: precision, recall, F1
> * evaluate the model while training: use the tensorboard

Thank you for your reply!
I tried with both 3 classes and one class. Both work, but its hard to tell which work best using my eyes as evaluation metrics. According to the code I'm using for evaluation, the IoU is 1.

Why would you suggest those metrics? From what I read the only thing important is IoU. I'm using this for my thesis. I think the more evaluation metrics the better.

"evaluate the model while training: use the tensorboard". I have never tried that, sounds like a good idea. Do you know any guide on how to implement it? 
I'm using a copy of this colab to run my code: https://colab.research.google.com/drive/1LIhBcxF6TUQUQCMEXCRBuF4a7ycUmjuw?usp=sharing#scrollTo=44999EylW2xL

**Comment by nataliameira** [comment]
@SaadShi 

I suggested these metrics because I found several papers using them.

I implemented the tensorboard on my notebook, this is the code:

import datetime
%load_ext tensorboard
%tensorboard –logdir logs/ --port=8008

to perform this you must be in the root directory.

**Comment by nataliameira** [comment]
@SaadShi 

You can change the hyperparameters of the rcnn mask implementation and view the metrics. You can be this guide: https://medium.com/analytics-vidhya/taming-the-hyper-parameters-of-mask-rcnn-3742cb3f0e1b

**Comment by SaadShi** [comment]
Oh, thank you! I'm also new to this, so the help is very appreciated!
I'm going to try
